.github/workflows/build-python-client.yml

```diff
@@ -0,0 +1,98 @@
+name: Python-client
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'python-client/**'
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # Do not stop when any job fails
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10" ]
+        os: [ubuntu-latest]
+        experimental: [false]
+        # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
+        include:
+          # - python-version: "3.11"
+          #   os: ubuntu-latest
+          #   experimental: true
+          # - python-version: "3.10"
+          #   os: macos-latest
+          #   experimental: true
+          # - python-version: "3.10" # https://github.com/tensorflow/io Does not talk about win 2022
+          #   os: windows-2022
+          #   experimental: true
+          - python-version: "3.10"
+            os: windows-2019
+            experimental: false
+    continue-on-error: ${{ matrix.experimental }} # https://ncorti.com/blog/howto-github-actions-build-matrix
+    steps:
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_OUTPUT
+        id: extract_branch
+
+      - name: Checkout code
+        uses: actions/checkout@v3.3.0
+
+      - name: Cache Python deps
+        uses: actions/cache@v3
+        with:
+          path: python-client/.venv
+          key: ${{ matrix.os }}-${{ matrix.python-version }}-python-deps-${{ hashFiles('python-client/tests/fixtures/**/*py')}}
+          restore-keys: ${{ matrix.os }}-${{ matrix.python-version }}-python-deps
+
+      - name: Setup PDM
+        uses: pdm-project/setup-pdm@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: false
```

Comment:
should we activate the caching of pdm installation here?

Reply:
ok, could you create a linear card to set the flag to `true` and reference this card it in a comment so that we don't forget

Reply:
Done : https://linear.app/giskard/issue/GSK-1712/activate-cache-on-pdm-setup-in-github-action